report_custom_filename/controllers/report_controller.py

```diff
@@ -38,18 +39,12 @@ def report_routes(self, reportname, docids=None, converter=None, **data):
         report_ids = report_xml.search(
             [('report_name', '=', reportname)])
         for report in report_xml.browse(report_ids):
-            if not report.download_filename:
+            if not report.download_filename or len(docids) != 1:
                 continue
             objects = http.request.session.model(report.model)\
                 .browse(docids or [])
-            generated_filename = email_template.mako_template_env\
-                .from_string(report.download_filename)\
-                .render({
-                    'objects': objects,
-                    'o': objects[:1],
-                    'object': objects[:1],
-                    'ext': report.report_type.replace('qweb-', ''),
-                })
+            generated_filename = safe_eval(report.download_filename,
```

Comment:
safe_eval uses Python syntax, not Jinja2 syntax, so it changes from the way Odoo parses this in the attachments.